src/thor/bidirectional_astar.cc

```diff
@@ -760,6 +783,11 @@ bool BidirectionalAStar::SetForwardConnection(GraphReader& graphreader, const BD
   EdgeStatusInfo oppedgestatus = edgestatus_reverse_.Get(oppedge);
   auto opp_pred = edgelabels_reverse_[oppedgestatus.index()];
 
+  // Disallow connections that are part of an uturn on an internal edge
+  if (pred.internal_turn() != InternalTurn::kNoTurn) {
```

Comment:
we can't allow connections here as the full uturn needs to be checked; otherwise, an edge case like a short route would be allowed.

Reply:
Forgive my ignorance here, but does this impact every travel mode and if so, is that desired? For example, could this negatively impact pedestrian routes?

Reply:
@browndp08 nope.  I noted it in the change log.  `Penalize pencil point uturns and uturns at short internal edges. Note: `motorcycle` and `motor_scooter` models do not penalize on short internal edges. No new uturn penalty logic has been added to the pedestrian and bicycle costing models.`

Reply:
this seems strange. why would we totally disallow this rather than penalize it? think of the flip side of the problem where the next opportunity to get to your destination is way way way down the road. uturns are bad for sure, but they arent as bad as driving an extra mile to make a loop you know what i mean?

Reply:
oh and also i get that there is a special consideration that its an internal edge but im not sure that that really matters. maybe you could explain a bit more why this is so undesirable that we would never want to suggest it

Reply:
hey @kevinkreiser
I do just penalize it.  Not sure why you thought I don't.  We just can't allow the connection there if is is not equal to `InternalTurn::kNoTurn`  We have to evaluate the full uturn

Reply:
oh im sorry i didnt pay attention to the precise wording there. you are saying we dont allow connecting there because we we cant consider both sides of the internal intersection (since it would be both search paths). i misunderstood that you were saying with the picture that we should never do a uturn over internal intersections. sorry about that

Reply:
oh no.  If we don't do this then that edge case could happen.

Reply:
yep i got ya, sorry for the confusion